gcp_variant_transforms/libs/processed_variant.py

```diff
@@ -343,6 +319,101 @@ def info_is_in_alt_bases(self, info_field_name):
+    has been initialized with valid `AlternateBaseData` objects. One instance
+    of this class should be created for each annotation field of each variant.
+    """
+    self._proc_var = proc_var
+    self._anntation_field_name = annotation_field_name
+    self._counter = counter
+    self._mismatch_counter = mismatch_counter
+    self._common_alt_ref_prefix = self._find_common_alt_ref_prefix()
+
+  def _find_common_alt_ref_prefix(self):
+    # tyep: () -> str
+    alt_list = [
+        alt.alternate_bases for alt in self._proc_var.alternate_data_list]
+    alt_list.append(self._proc_var.reference_bases)
+    return os.path.commonprefix(alt_list)
+
+  def add_annotation_data(self, annotation_names, data):
+    # type: (List[str], List[str]) -> None
+    """The main function for adding annotation data to `self._proc_var`."""
+    alt_annotation_map = self._convert_annotation_strs_to_alt_map(
+        annotation_names, data)
+    for alt_bases, annotations_list in alt_annotation_map.iteritems():
+      self._add_annotation_list(alt_bases, annotations_list)
+
+  def _convert_annotation_strs_to_alt_map(self, annotation_names, field_data):
+    # type: (List[str], List[str]) -> Dict[str, List[Dict[str, str]]]
+    alt_annotation_map = defaultdict(list)
+    for annotation_str in field_data:
+      annotations = _extract_annotation_list_with_alt(annotation_str)
+      alt_annotation_map[annotations[0]].append(
+          self._create_annotation_map(annotations, annotation_names))
+    return alt_annotation_map
+
+  def _create_annotation_map(self, annotations, annotation_names):
+    # type: (List[str], List[str]) -> Dict[str, str]
+    if len(annotation_names) != len(annotations) - 1:
+      raise ValueError('Expected {} annotations, got {}'.format(
+          len(annotation_names), len(annotations) - 1))
+    annotation_dict = {}
+    for index, name in enumerate(annotation_names):
+      annotation_dict[name] = annotations[index + 1]
+    return annotation_dict
+
+  def _add_annotation_list(self, alt_bases, annotations_list):
+    # type:
+    # This assumes that number of alternate bases and annotation segments
+    # are not too big. If this assumption is not true, we should replace the
+    # following loop with a hash table search and avoid the quadratic time.
+    for alt in self._proc_var._alternate_datas:
+      if self._alt_matches_annotation_alt(alt.alternate_bases, alt_bases):
+        alt._info[self._anntation_field_name] = annotations_list
+        self._counter.inc()
+        break
+    else:
+      self._mismatch_counter.inc()
+      logging.warning(
+          'Could not find matching alternate bases for %s in '
+          'annotation filed %s', alt_bases, self._anntation_field_name)
+
+  def _alt_matches_annotation_alt(self, alt_bases, annotation_alt):
+    # type: (str, str) -> bool
+    """Returns true if `alt_bases` matches`annotation_alt`
+    See: https://useast.ensembl.org/info/docs/tools/vep/vep_formats.html
```

Comment:
Please clarify which section in the doc talks about this logic. Also, please summarize the logic here as well (links can become broken over time) and provide a few examples.